pkg/bpf/c/restricted-network.bpf.c

```diff
@@ -70,13 +88,51 @@ static inline void report_ipv4_event(void *ctx, u64 cg, enum action action, enum
   bpf_ringbuf_output(&audit_events, &ev, sizeof(ev), 0);
 }
 
+static inline void report_ipv6_event(void *ctx, u64 cg, enum action action, enum lsm_hook_point point, struct socket *sock, const struct sockaddr_in6 *daddr)
+{
+  struct audit_event_ipv6 ev;
+  
+  struct task_struct *current_task;
+  struct uts_namespace *uts_ns;
+  struct nsproxy *nsproxy;
+  current_task = (struct task_struct *)bpf_get_current_task();
+  
+  bpf_core_read(&nsproxy, sizeof(nsproxy), &current_task->nsproxy);
```

Comment:
bpf_core_read と BPF_CORE_READ ってどう違うんですっけ...

Reply:
> With vanilla bpf_probe_read()/bpf_core_read(), this will turn into 4 calls with an extra temporary variable to store value for all those intermediate pointers

ref https://facebookmicrosites.github.io/bpf/blog/2020/02/19/bpf-portability-and-co-re.html

とのことなので、原則小文字の `bpf_core_read` は使う必要がなくて、 `BPF_CORE_READ` に書き換えて良さそうに思いました。
BPFではレジスタが貴重なのでそうした方がトラブルが減りそうな予感がします。

Reply:
特定のアドレスを更新したい場合は `BPF_CORE_READ_INTO()` が使えそう